### ISSUE TITLE
Autosave to temporary file

### ISSUE THREAD

**Issue by brandonzylstra** [atom/autosave#11]
It would be extremely useful if this gave the option to save to a temporary file rather than the regular file, since overwriting the on-disk version of a file is not always desired, but preserving a copy of all changes made in memory is always useful.

**Comment by jachinte** [comment]
+1

**Comment by lambdafu** [comment]
I gave atom.io a try the other day and lost all pending edits after a system crash. Autosaving to temporary files is a must have feature, so I am back to good ol' emacs.

**Comment by Poetro** [comment]
+1

**Comment by sambao21** [comment]
:+1:

**Comment by mosauter** [comment]
+1

**Comment by lcb931023** [comment]
+myAxe

**Comment by teheavy** [comment]
+1

**Comment by Meligy** [comment]
+1 .. Some 4rd party tried to do it in [save-session](https://github.com/mpeterson2/save-session) but it does need an insider to make it integrate properly, for example, it doesn't detect the different between tab close (no save required), and entire app close (save should hppen), as in [save-session#121](https://github.com/mpeterson2/save-session/issues/121)

**Comment by s7v7nislands** [comment]
+1

**Comment by asbjornu** [comment]
This is the number one gripe I have with Atom and the feature I miss most from Sublime Text. :+1:

**Comment by zenna** [comment]
+1

**Comment by stilliard** [comment]
+1 this would be awesome.

**Comment by wbecker** [comment]
Notable changes in [1.4.0](https://github.com/atom/atom/releases/tag/v1.4.0) says that "Untitled documents in a project are now serialized and restored." 

Does this fix / relate to this issue?

**Comment by asbjornu** [comment]
@wbecker It covers my needs perfectly, at least. Thanks for pointing it out! :+1:

**Comment by Meligy** [comment]
Does "in a project" mean anything specific?

**Comment by lee-dohm** [comment]
@Meligy Any time you open Atom you are implicitly in a project. A project is implicitly defined based on the list of root folders. So if you do `atom ~/foo` that is one project. Then later if you open a new session of atom with `atom ~/bar` that is a different project.

**Comment by forestsheep** [comment]
+65535

**Comment by Meligy** [comment]
Well, it's not working for me. It seems to work when I have a folder open in sidebar, but if I remove last folder from project, restart, and try to lave a file open and restart again, it's gone.

I have tried that with latest Atom beta (1.5 beta) on latest Mac, and with Atom release (1.4) on Windows 10. Same behaviour.

I think maybe it only saves unsaved changes to already (previously) saved files, but the file was never saved (never existed on disk or had a location path), it doesn't restore it. And the option to save temporary version to disk before saving files doesn't seem to change this.

**Comment by lee-dohm** [comment]
@Meligy you're right, we're still working on preserving the state of untitled buffers. You can find links to all the current stuff about them here https://github.com/atom/atom/pull/9968.

**Comment by denieler** [comment]
We are supporting your idea with [save-workspace](https://atom.io/packages/save-workspace) package. Where we are saving whole workspace with project. Please look at out project and we will be appreciate if you'll get us valuable suggestions and, sure, welcome to contribute! 😄

**Comment by jessejiang0214** [comment]
Need this feature, I thought it like Sublime Text, it works fine at begin. But after few days working, I lost all files in Atom

**Comment by jeffmcneill** [comment]
Pretty sure https://github.com/atom/autosave does the trick these days. If you open atom in a given directory, then changes to that set of tabs should be saved, along with unsaved tabs as well. That said, a system crash would still wipe out the unsaved tabs, so best to "name and save" each tab, including any temporary ones. I also like https://atom.io/packages/file-watcher for rare file conflicts.

**Comment by phord** [comment]
@jeffmcneill Since version 1.4, Atom keeps "untitled" windows just fine even when there's a crash.  It does this even if autosave is disabled.  Autosave is used to save the contents to the original file instead of to a hidden Atom-state-cache.

**Comment by asbjornu** [comment]
@phord: Does that work with or without any projects?

**Comment by phord** [comment]
@asbjornu It works with projects.  It does not work without a project.  Also, I didn't realize this issue was about the autosave plugin when I replied; I thought I was replying to an issue on atom/atom.

Here's something funny:  If you open atom without a project, open an unnamed file, type something into it, and then exit Atom, you get this dialog:

'untitled' has changes, do you want to save them?
Your changes will be lost if you close this item without saving.
![image](https://user-images.githubusercontent.com/123908/26942839-6599d298-4c38-11e7-8c73-9dea578ca4de.png)

Here's what the buttons do:
Save: A save dialog appears so you can name the file and save it. [*]
Cancel: No save dialog appears and Atom exits immediately
Don't save: No save dialog appears and Atom exits immediately

[*] If you choose "Save" but then don't actually save the file (because you canceled the Save dialog), Atom exits immediately.

As far as I can tell, there is no difference between the "Cancel" and "Don't Save" buttons.  For that matter, "Save + Cancel" also is the same.

In all cases, when you reopen Atom without a project, you will have an 'untitled' file open again with no contents.

**Comment by g3blv** [comment]
It is not quite clear to me if unsaved filed are (auto) saved or not saved as temporary files.
